### PR TITLE
[Handshake] Refactor handshake runner to be visitor based

### PR DESCRIPTION
**Pull request by mortbopet**
This commit refactors the handshake-runner to be based on a visitor pattern rather than manually checking for the type of each possible operation. The commit also takes a step towards reducing the code duplication between executing a mlir::FuncOp and handshake::FuncOp.

More opportunities for cleanup and refactoring exists, however, this commit focuses only on the visitor refactoring to keep it as brief as possible.

### CODE REVIEW

File: tools/handshake-runner/Simulation.cpp [hunk 1]
```diff
@@ -45,127 +51,298 @@ static void fatalValueError(StringRef reason, T &value) {
...
+    int64_t width = INDEX_WIDTH;
+    APInt aparg(width, x);
+    return aparg;
+  } else if (type.isa<mlir::IntegerType>()) {
+    int64_t x;
+    arg >> x;
+    int64_t width = type.getIntOrFloatBitWidth();
+    APInt aparg(width, x);
+    return aparg;
+    // } else if (type.isF16()) {
```
Comment (mikeurbach) [inline tools/handshake-runner/Simulation.cpp:98]: remove commented out code, or uncomment?

File: tools/handshake-runner/Simulation.cpp [hunk 2]
```diff
@@ -195,306 +377,134 @@ void executeOp(mlir::memref::StoreOp op, std::vector<Any> &in,
...
-    llvm_unreachable("unknown type");
+void HandshakeExecuter::execute(mlir::BranchOp branchOp, std::vector<Any> &in,
+                                std::vector<Any> &) {
+  mlir::Block *dest = branchOp.getDest();
+  unsigned arg = 0;
+  for (mlir::Value out : dest->getArguments()) {
+    LLVM_DEBUG(debugArg("ARG", out, in[arg], time));
+    valueMap[out] = in[arg];
+    timeMap[out] = time;
+    arg++;
```
Comment (mikeurbach) [inline tools/handshake-runner/Simulation.cpp:402]: nit: i think pre-increment is preferred.

**Review by mikeurbach — APPROVED** [review]
I'm not super familiar with this, but the refactor makes sense to me. Just a couple minor nits.

File: tools/handshake-runner/Simulation.cpp (hunk 1 above)
Comment (mortbopet) [reply tools/handshake-runner/Simulation.cpp:98]: Will do; there seems to be quite a bit of lingering commented out pieces of code around these parts...

**Pull request merged**